pkg/test/nodes.go

```diff
@@ -62,3 +64,18 @@ func Node(overrides ...NodeOptions) *v1.Node {
 		},
 	}
 }
+
+func MachineLinkedNode(machine *v1alpha5.Machine) *v1.Node {
```

Comment:
Doesn't this need to return the machine as the node's owner?

Reply:
We unfortunately can't do this in this function since you need to know the UUID in order to add an owner reference and this isn't resolved until after the object is created. Realistically, this doesn't really matter for testing since there is no controller-manager that is monitoring the owner references anyways